File: memory-api/src/main/java/com/memory/service/calendar/factory/CalendarEventFactoryService.java
package com.memory.service.calendar.factory;

import com.memory.dto.calendar.CalendarEventRequest;
import com.memory.dto.calendar.response.BaseCalendarEventResponse;

public interface CalendarEventFactoryService {

    BaseCalendarEventResponse createCalendarEvent(Long memberId, CalendarEventRequest.Create request);

}

File: memory-api/src/main/kotlin/com/memory/service/calendar/factory/CalendarEventFactoryService.kt
package com.memory.service.calendar.factory;

import com.memory.dto.calendar.CalendarEventRequest;
import com.memory.dto.calendar.response.BaseCalendarEventResponse;

import java.util.List;
import java.time.LocalDateTime;

public interface CalendarEventFactoryService {

    BaseCalendarEventResponse createCalendarEvent(Long memberId, CalendarEventRequest.Create request);

    BaseCalendarEventResponse updateCalendarEvent(Long memberId, Long eventId, CalendarEventRequest.Update request);

    List<BaseCalendarEventResponse> getCalendarEventsByDateRange(Long memberId, LocalDateTime startDate, LocalDateTime endDate);

    List<BaseCalendarEventResponse> getCalendarEventsWithDday(Long memberId);
}
